lib/node_modules/@stdlib/array/base/reshape/lib/main.js

```diff
@@ -0,0 +1,99 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var flatten = require( '@stdlib/array/base/flatten' );
+var slice = require( '@stdlib/array/base/slice' );
+
+
+// FUNCTIONS //
+
+/**
+* Recursive reshapes an array.
+*
+* @private
+* @param {Array} array - input n-dimensional nested array
+* @param {NonNegativeInteger} ndims - number of dimensions
+* @param {NonNegativeIntegerArray} shape - array shape
+* @param {NonNegativeInteger} dim - dimension index
+* @param {NonNegativeInteger} index - sub-array index
+* @returns {Array} output m-dimensional nested array
+*/
+function recurse( array, ndims, shape, dim, index ) {
+	var stepSize;
+	var subArray;
+	var out;
+	var d;
+	var S;
+	var i;
+	var j;
+
+	S = shape[ dim ];
+	if ( dim === ndims - 1 ) {
+		return slice( array, index, index + S );
+	}
+
+	d = dim + 1;
+	stepSize = 1;
+	for ( i = d; i < shape.length; i++ ) {
+		stepSize *= shape[ i ];
+	}
+
+	out = [];
+	for ( j = 0; j < S; j++ ) {
```

Comment:
Why do you need a separate variable `j` and you've already got `i` which is used above?